packages/leafygreen-provider/src/ModalPopoverContext/ModalPopoverContext.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { act, render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { renderHook } from '@leafygreen-ui/testing-lib';
+
+import {
+  ModalPopoverProvider,
+  useModalPopoverContext,
+} from './ModalPopoverContext';
+import { type ModalPopoverProviderProps } from './ModalPopoverContext.types';
+
+const childTestID = 'modal-popover-provider';
+const buttonTestId = 'test-button';
+
+function TestContextComponent() {
+  const { isPopoverOpen, setIsPopoverOpen } = useModalPopoverContext();
+
+  return (
+    <>
+      <div data-testid={childTestID}>
+        {isPopoverOpen !== undefined ? isPopoverOpen.toString() : ''}
+      </div>
+      <button
+        onClick={() => setIsPopoverOpen(true)}
+        data-testid={buttonTestId}
+      />
+    </>
+  );
+}
+
+function renderProvider(props?: ModalPopoverProviderProps) {
+  const utils = render(
+    <ModalPopoverProvider {...props}>
+      <TestContextComponent />
+    </ModalPopoverProvider>,
+  );
+  const testChild = utils.getByTestId(childTestID);
+  return { ...utils, testChild };
+}
+
+describe('packages/leafygreen-provider/ModalPopoverContext', () => {
+  test('only renders children in the DOM', () => {
+    const { container, testChild } = renderProvider();
+    expect(container.firstChild).toBe(testChild);
+  });
+
+  test('`isPopoverOpen` is initialized as false', () => {
+    const { testChild } = renderProvider();
+    expect(testChild.textContent).toBe('false');
+  });
+
+  test('when passed true, `setIsPopoverOpen` sets `isPopoverOpen` to true', () => {
+    const { testChild, getByTestId } = renderProvider();
+
+    // The button's click handler fires setIsPopoverOpen(true)
+    userEvent.click(getByTestId(buttonTestId));
+
+    expect(testChild.textContent).toBe('true');
+  });
```

Comment:
Technically we don't need these tests, since we test `useModalPopoverContext` explicitly below